frontend/src/components/Table/Table.tsx

```diff
@@ -234,25 +242,31 @@ const Table = <T extends object>(props: PropsWithChildren<TableProps<T>>): React
   }, [data, setSelectedRows, page, selectedRowIds]);
 
   const getNextSortDirection = (column: ColumnInstanceWithProps<T>): SortDirection => {
-    if (!props.sort) return 'asc';
+    if (!(props.sort as any)[column.id]) return 'asc';
 
-    if (props.sort.column !== column.id || props.sort.direction === 'desc') {
-      return 'asc';
+    if ((props.sort as any)[column.id] === 'desc') {
+      return undefined;
```

Comment:
'desc' results in undefined?

Reply:
nvm, I understand, this is a three state toggle switch between asc,desc and undefined with no way of going directly from desc to asc.

Reply:
Yes, thats what it represents. `undefined` => `asc`, `asc` => `desc`, `desc` => undefined